test/assets/unrunnableTestCases.csv

```diff
@@ -4079,6 +4070,7 @@ XQST0089,AssertionError: expected [Function] to throw an error
 XQST0093a,Error: Duplicate function registration
 XQST0125_1,AssertionError: expected [Function] to throw an error
 XQST0125_2,AssertionError: expected [Function] to throw an error
+typeswitch-in-xpath,AssertionError: expected [Function] to throw an error
```

Comment:
This error seems to be new, can we investigate this and fix if possible?

Reply:
I think the test here tests whether you can use the typeswitch expression in XPath. You shouldn't be able to. Could you add an error to the compiler like how for example the element constructors throw errors in the compiler?

Reply:
The tests tries to execute a typeswitch in xPath and expects an error, but adding the `!compilationOptions.allowXQuery` validation is not working, as it's never satisfied, even in that test. I don't know if there's another way to do that, but right now, that error is the expected behaviour.

Reply:
Yes, we checked and found the issue. To run the QT3 tests we [determine the language](https://github.com/FontoXML/fontoxpath/blob/master/test/qt3tests.ts#L472) based on the test file / test case.
A debug session indeed shows we use XQuery for this test case instead of XPath. The issue is that the language to use is either a descendant of the parent or of descendant of the test case.

We should always look at the descendants of the test case first before checking the descendants of the parent.

Could you fix that in this PR or make an issue for this so we can fix it later? Either way is ok with me, just would like to see this fixed ;)

Reply:
I did check this with the debug page and it does indeed throw the error which is correct behaviour.